SimCalorimetry/EcalSimAlgos/src/EcalTimeMapDigitizer.cc

```diff
@@ -99,17 +102,39 @@ void EcalTimeMapDigitizer::add(const std::vector<PCaloHit>& hits, int bunchCross
 
       TimeSamples& result(*findSignal(detId));
 
+    if (nullptr != m_ComponentShapes) {
+      // for now we have waveform_granularity = 1., 10 BX, and waveform capacity 250 -- we want to start at 25*bunchCrossing and go to the end of waveform capacity
+      double binTime(0);
+      for(unsigned int bin(0); bin != result.waveform_capacity; ++bin) {
+        if (ComponentShapeCollection::toDepthBin((*it).depth()) <= ComponentShapeCollection::maxDepthBin()) {
+          result.waveform[bin] += (*(shapes()->at((*it).depth())))(binTime-jitter-25*(bunchCrossing-m_minBunch))* (*it).energy();
+        }
+        #ifdef waveform_debug
+        else { 
+          std::cout << "strange depth found: " << ComponentShapeCollection::toDepthBin((*it).depth()) << std::endl;
+        } // note: understand what these depths mean
+        #endif
+        binTime += result.waveform_granularity;
+      }
+    }
+
       //here fill the result for the given bunch crossing
+
+      // i think this is obsolete now that there is a real MTD
```

Comment:
Is it?

Reply:
Good question. I am not sure if anyone uses this part of the code any more (from archaeological digging it appears that the EcalTimeMapDigitizer was implemented in the days of early studies of the MTD, and to model a thin layer detector this code only considers hits in a certain 1cm deep layer of ECAL. but now if one wants to do MTD studies one can use the actual MTD simulation.) I made this change to let it consider all hits in the ECAL crystals since I think if one were to use this code for ECAL studies then it makes more sense to consider all the depths in ECAL instead of one thin slice.

I realize this is a big departure in the behavior of what this code will do, but I think it's an improvement; however if anyone complains I am happy to revert it to the original behavior.